controllers/component_build_controller_common.go

```diff
@@ -305,3 +305,26 @@ func getPathContext(gitContext, dockerfileContext string) string {
 	path = strings.TrimPrefix(path, separator)
 	return path
 }
+
+func getPipelineNameAndBundle(pipelineRef *tektonapi.PipelineRef) (string, string, error) {
+	if pipelineRef.Resolver != "" && pipelineRef.Resolver != "bundles" {
+		return "", "", boerrors.NewBuildOpError(
+			boerrors.EUnsupportedPipelineRef,
+			fmt.Errorf("unsupported Tekton resolver %q", pipelineRef.Resolver),
+		)
+	}
+
+	name := pipelineRef.Name
+	var bundle string
+
+	for _, param := range pipelineRef.Params {
+		switch param.Name {
+		case "name":
+			name = param.Value.StringVal
+		case "bundle":
+			bundle = param.Value.StringVal
+		}
+	}
+
```

Comment:
since bundle can be empty

I'd return permanent error here when bundle is empty